modules/consensus/src/lib.rs

```diff
@@ -0,0 +1,172 @@
+use futures::channel::mpsc::SendError as FuturesSendError;
+use futures::Sink;
+use futures::SinkExt;
+use futures_util::StreamExt;
+use solana_lite_rpc_core::block_store::BlockInformation;
+use solana_sdk::epoch_info::EpochInfo;
+use solana_transaction_status::EncodedConfirmedBlock;
+use std::collections::HashMap;
+use thiserror::Error;
+use tokio::pin;
+use tokio::task::JoinHandle;
+use yellowstone_grpc_client::GeyserGrpcClient;
+use yellowstone_grpc_client::GeyserGrpcClientError;
+use yellowstone_grpc_proto::geyser::CommitmentLevel;
+use yellowstone_grpc_proto::prelude::SubscribeRequestFilterAccounts;
+use yellowstone_grpc_proto::prelude::SubscribeRequestFilterBlocks;
+use yellowstone_grpc_proto::prelude::{subscribe_update::UpdateOneof, SubscribeRequestFilterSlots};
+
+#[derive(Debug, Error)]
+pub enum HistoryError {
```

Comment:
IMO that type should go into _history_ module

Reply:
Update done